### DISCUSSION TITLE
Can we compile the policy bundle into wasm at the OPA server end

### DISCUSSION THREAD

**Discussion by dushyant7917** [open-policy-agent/community#362] [Wasm]
If OPA server uses poll model, it fetches the policy bundles from a remote server. If the policy bundle doesn't contain the .wasm file the server compiles the policy at server end and uses rego. But if the policy bundle contains .wasm file the OPA server doesn't need to compile the policy as .wasm file is already a compiled file and it uses that to make the decision.
Is there a way of creating .wasm at the OPA server end after the policy bundle has been fetched from the remote server as usual?

**Comment by srenatus** [comment]
Hey there.

That's certainly an interesting idea, and it would be possible to do with OPA's means. But it has never come up before that someone wanted to do that (to my knowledge). You could file an issue, as an enhancement request, I suppose.

That said, what's the rationale behind this?

Also, I think it should be possible to create some sort of JIT proxy that downloads rego bundles, compiles them to wasm bundles, and hands out the wasm bundles. 🤔

**Reply by dushyant7917** [reply]
So for compiling the policy into wasm we need to provide the entry point parameter (`-e`). What if the entry point is dynamic and decided by the client contacting the OPA server for decision. In that case we can't compile the policy into wasm in bundle server because it doesn't have the dynamic entry point.
